Wikipedia/Code/ReadingThemesControlsPresenter.swift

```diff
@@ -0,0 +1,130 @@
+        
+        readingThemesControlsViewController.delegate = self
+        readingThemesControlsViewController.setValuesWithSteps(fontSizes.count, current: index)
+        readingThemesControlsViewController.apply(theme: theme)
+        
+        let popoverPresentationController = readingThemesControlsViewController.popoverPresentationController
+        
+        popoverPresentationController?.barButtonItem = readingThemesControlsToolbarItem
+        popoverPresentationController?.permittedArrowDirections = [.down, .up]
+        popoverPresentationController?.backgroundColor = theme.colors.popoverBackground
+        viewController.present(readingThemesControlsViewController, animated: true, completion: nil)
+        
+        if let navBar = viewController.navigationController?.navigationBar,
+            passthroughNavBar {
+            popoverPresentationController?.passthroughViews = [navBar]
+        }
+    }
+    
+    func dismissReadingThemesPopoverIfActive(from viewController: UIViewController) {
+        if viewController.presentedViewController is ReadingThemesControlsViewController {
+            viewController.dismiss(animated: true, completion: nil)
+        }
+    }
+    
+    //MARK: WMFReadingThemesControlsViewControllerDelegate
+    
+    func fontSizeSliderValueChangedInController(_ controller: ReadingThemesControlsViewController, value: Int) {
+        let fontSizes = fontSizeMultipliers
+        
+        if value > fontSizes.count {
+            return
+        }
+        
+        let multiplier = fontSizeMultipliers[value]
+        let nsNumber = NSNumber(value: multiplier)
+        wkWebView.wmf_setTextSize(multiplier)
+        UserDefaults.wmf.wmf_setArticleFontSizeMultiplier(nsNumber)
+    }
+}
+
+//objective-c wrapper for Article presentation.
+@objc(WMFReadingThemesControlsPresenter)
+class ReadingThemesControlsPresenter: NSObject, ReadingThemesControlsPresenterProtocol {
+    
+    var passthroughNavBar: Bool {
+        return true
+    }
+    
+    var readingThemesControlsViewController: ReadingThemesControlsViewController
+    
+    var wkWebView: WKWebView
+    
+    var readingThemesControlsToolbarItem: UIBarButtonItem
+    
+    @objc var objcIndexOfCurrentFontSize: Int {
+        return indexOfCurrentFontSize
+    }
+    
+    @objc var objcFontSizeMultipliers: [Int] {
+        return fontSizeMultipliers
+    }
+    
+    @objc init(readingThemesControlsViewController: ReadingThemesControlsViewController, wkWebView: WKWebView, readingThemesControlsToolbarItem: UIBarButtonItem) {
+        self.readingThemesControlsViewController = readingThemesControlsViewController
+        self.wkWebView = wkWebView
+        self.readingThemesControlsToolbarItem = readingThemesControlsToolbarItem
+        super.init()
+    }
+    
+    @objc func objCShowReadingThemesControlsPopup(on viewController: UIViewController, theme: Theme) {
```

Comment:
I'm not crazy about these wrapper methods - definitely open to suggestions on this.

Reply:
I think this is a fine workaround for now. The `mobile-html` work will likely involve re-writing `WMFArticleViewController` in Swift which would eliminate the need for this